### DISCUSSION TITLE
Adding --archive-flags="-destination 'generic/platform=iOS'" to flutter project

### DISCUSSION THREAD

**Discussion by dqr79** [codemagic-ci-cd/codemagic-docs#971] [Q&A] [answered]
My flutter build for iOS just failed with this error :

error: Provisioning profile "XXXX" doesn't include the currently selected
device "builder's Virtual Machine" (identifier XXXXXXXX-XXXXXXXXXXXXXXXX).
(in target 'XXXXXXX' from project 'App').

In the docs, I found that the solution is to add the following to the end of m my xcode-project build-ipa command:
--archive-flags="-destination 'generic/platform=iOS'"

I couldn't figure out how to do this in the workflow editor which I've been using so far - so I tried my luck with codemagic.yaml

Of course, this being a flutter project, it doesn't include the xcode-project build-ipa command, it uses flutter build ipa instead. But there are no instructions on how to solve this error in that case, so I tried my luck adding the  same flag;
 flutter build ipa --export-options-plist=~/export_options.plist  --archive-flags="-destination 'generic/platform=iOS'"

..but of course, it did not work. Any ideas on how to resolve this error?

**Comment by himesh-cm** [comment, marked as answer]
Hi @dqr79 can you please share the build id?

**Reply by dqr79** [reply]
Hi himesh, sure it's 632af0177323ea502620bf15

**Reply by himesh-cm** [reply]
Thank you, i will check and get back to you.

**Reply by himesh-cm** [reply]
Hi @dqr79 
Since you are using the yaml configuration, you can follow the below commands.

```
cd ios
xcode-project build-ipa --archive-flags="-destination 'generic/platform=iOS'"
```

**Reply by dqr79** [reply]
Thanks himesh, but it did not work either, please see build id 632c36f73dad27f72abcd0d8
it says 
Build failed :|
Step 9 script `xcode-project build-ipa --archive-flags="-destination 'generic/platform=iOS'"` exited with status code 2

Am I missing something here? Here's my related scripts block:
      - flutter packages pub get
      - find . -name "Podfile" -execdir pod install \;
      - keychain initialize
      - app-store-connect fetch-signing-files "com.my.projectname" --type IOS_APP_DEVELOPMENT
        --create
      - keychain add-certificates
      - xcode-project use-profiles
      - cd ios
      - xcode-project build-ipa --archive-flags="-destination 'generic/platform=iOS'" 


By the way, if this did end up working eventually, will I be able to add the flags I was using for flutter build ipa? eg --export-options-plist=~/export_options.plist or various --dart-define values? Because I do need those for my build..
THank you!!

**Reply by himesh-cm** [reply]
Hi @dqr79 

Can you please share the screenshot of your Xcode after opening the iOS directory?
Please select your target.

For Reference
![image](https://user-images.githubusercontent.com/106069365/191756058-f0d5fb29-20a6-4a37-872f-dd549089b745.png)

**Reply by dqr79** [reply]
hey himesh, 
here you go
![Screenshot 2022-09-22 at 22 07 50](https://user-images.githubusercontent.com/12547229/191843455-56589e23-6d7c-4b29-97ac-f5b97f7fbb2c.png)

Thanks again for your help and effort!

**Reply by himesh-cm** [reply]
Hi @dqr79 
Can you please check in `ios/Runner.xcodeproj/project.pbxproj` whether `SUPPORTS_MACCATALYST` is set to YES or NO.

awaiting your feedback.

**Reply by himesh-cm** [reply]
Hi @dqr79 - were you able to look into this issue?

**Reply by dqr79** [reply]
Hi @himesh-cm sorry about the delay - I just SUPPORTS_MACCATALYST key in that file for me. Should I add it, and if so, where? My xcode is an older version so that may be the reason for not having this key -  and I'm not able to update it for technical reasons right now (I'd have to replace my hdd with a bigger one, or replace my MacBook..) And I'd just quickly like to run this one build because I had to add an additional tester's phone.. So far I had no problem running the builds with Codemagic, I hope there's some workaround without having to update my xcode right now, because of this one thing.. Thank you!

**Reply by himesh-cm** [reply]
Can you share what macOS version and Xcode version you are using locally?

**Reply by dqr79** [reply]
Mac OSX: 10.15.7, Xcode: 11.7

**Reply by dqr79** [reply]
@himesh-cm what's up, any ideas on what I could do as a workaround?

**Reply by himesh-cm** [reply]
Hi @dqr79 
Apologies, the thread got lost; on another note, the macOS and Xcode version is pretty old, and sadly I cannot replicate the issue.
We know that enabling SUPPORTS_MACCATALYST does resolve the issue, but it seems that the Xcode version you use locally does not support that.

can you please try if this helps?
```
cd ios
xcode-project build-ipa --archive-flags="-destination 'platform=macOS,variant=Mac Catalyst' 'generic/platform=iOS'"
```

**Reply by dqr79** [reply]
@himesh-cm  thank you, I tried, I got a brand new error now..
`xcode-project: error: argument --workspace: Workspace or project argument needs to be specified`
By the way, if we could get this to work eventually, will there be a way to use --dart-define flags with xcode-project build-ipa? Because my app uses dart define params, they are essential in my workfow but I just realized, xcode-project doesn't seem to support them - so even if you came up with a solution, or my macbook was eventually replaced, I wouldn't be able to use those flags.. So I'm wondering, if there is any way to go back to flutter build ipa, and resolve the original error message I started this discussion with? I mean it's a flutter project, so that would be best practice anyway I think..

**Reply by himesh-cm** [reply]
I am afraid not; the safest way is to build using the 13 / latest Xcode version locally and enable `SUPPORTS_MACCATALYST` support.

**Reply by dqr79** [reply]
@himesh-cm thanks for your help but I'm not sure I understand clearly - are you suggesting I stop using codemagic for my project because of this issue, and start building locally instad, as I won't be able to use flutter build ipa on codemagic anymore, because of some unresolved issue?

**Reply by himesh-cm** [reply]
Hi @dqr79 
I am not suggesting you stop using the service. It is always a good practice to have the latest software versions local and on CI to avoid breaking changes.

Projects configured on Xcode 11 may not work as expected when you compile them on newer versions. if you lower the Xcode version on CI to build that we support, you won't be able to support the latest iOS version.

**Reply by himesh-cm** [reply]
Another option would be to avoid using Adhoc Profile and use an App store profile, please check if that helps.

**Reply by himesh-cm** [reply]
Hi @dqr79, can you please confirm if you still face this issue?

**Reply by dqr79** [reply]
Hi @himesh-cm  yes I still face this issue, I'm currently stuck as I would have to update my Xcode but I'm not able to do it on my current Macbook, so I will have to come back to this once I have replaced it - which can easily be a month or two. Until then it seems like I will be stuck.

**Reply by dqr79** [reply]
Hi @himesh-cm  I tried it on a later macbook version of a friend, but he could not enable the Mac Catalyst, it was dimmed out - so we were trying to figure out why it is dimmed out, which we could not, but in the process I found that Mac Catalyst is for bringing existing iOS applications to macOS. This is not something we are planning to do and the app we are building is not built in a way that makes this possible so can you please explain why codemagic would force me to build this app with this support enabled? 
Are you entirely certain this is the solution to an error message, "Provisioning profile "XXXX" doesn't include the currently selected device "builder's Virtual Machine" which suggests that the provisioning profile should include the virtual machine the build is being executed on?

**Reply by himesh-cm** [reply]
Can you share the latest build id, so I can investigate the same?

**Reply by dqr79** [reply]
@himesh-cm  I'm happy to report that I've been able to resolve the issue in the meantime, without using Mac Catalyst support - I had to update target versions and podfile platforms and now the build went through without problems - so it's all good now.

**Reply by himesh-cm** [reply]
Can you share what changes you made in podifle?
It can be helpful to others
